commons/src/main/java/org/dhis2/commons/periods/domain/GetEventPeriods.kt

```diff
@@ -0,0 +1,60 @@
+package org.dhis2.commons.periods.domain
+
+import androidx.paging.Pager
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
+import androidx.paging.map
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import org.dhis2.commons.periods.data.EventPeriodRepository
+import org.dhis2.commons.periods.data.PeriodLabelProvider
+import org.dhis2.commons.periods.data.PeriodSource
+import org.dhis2.commons.periods.model.Period
+import org.hisp.dhis.android.core.period.PeriodType
+import org.hisp.dhis.android.core.program.ProgramStage
+import java.util.Date
+
+class GetEventPeriods(
+    private val eventPeriodRepository: EventPeriodRepository,
+    private val periodLabelProvider: PeriodLabelProvider = PeriodLabelProvider(),
+) {
+    fun fetchPeriods(
```

Comment:
could this be an invoke with the operator `modifier`, here there is an explanation why make sense use cases to be used with the invoke: https://developer.android.com/topic/architecture/domain-layer?authuser=1#use-cases-kotlin